cpp/benchmarks/search/search.cpp

```diff
@@ -24,49 +24,69 @@
 #include <cudf/sorting.hpp>
 #include <cudf/types.hpp>
 
-class Search : public cudf::benchmark {
+namespace {
+template <typename Type>
+std::unique_ptr<cudf::table> create_table_data(cudf::size_type n_rows,
```

Comment:
1. The existence of `create_table_data` and `create_column_data` is somewhat concerning to me. We have a set of APIs in `common/generate_input.hpp` that are used pretty much universally throughout our benchmarks thanks to hard work from @karthikeyann. Are helper functions really needed? The input assumptions of each benchmark are clearer when the profile and null frequency are specified inline, rather than hidden in a helper function.

2. Why does this PR refactor all of `upper_bound` and `lower_bound` benchmarks? Those changes seem unrelated to the goal of this PR (improving `cudf::contains`). If refactoring is required, it should be done separately so that @GregoryKimball's benchmarks are stable - we want to be able to point to a single commit during git bisection, and this alters too many benchmarks at once to be regarded as "atomic" in that regard.

3. This PR is introducing new benchmarks for `SearchScalar_AllValid` and `SearchScalar_Nulls`. Per #7960, we should write all new benchmarks in nvbench.

Reply:
I think what I want to see is the following:
- Revert changes to `cpp/benchmarks/search/search.cpp`
- Write a new nvbench file for the scalar contains benchmarks
- Make a follow-up PR to port the rest of the search benchmarks to nvbench

Some of the changes in this PR are genuinely helpful, but we need to keep changes as atomic as possible and do no more work than is needed in a single PR.